apps/src/templates/sectionProgress/VirtualizedDetailView.jsx

```diff
@@ -25,37 +25,51 @@ const styles = {
   }
 };
 
-const list = [
-  ["Lesson", "1", "2", "3", "4"],
-  ["Level Type", "1", "2", "3", "4"],
-  ["Student 1"],
-  ["Student 2"],
-  ["Student 3"],
-  ["Student 4"],
-  ["Student 5"],
-  ["Student 6"],
-  ["Student 7"],
-  ["Student 8"],
-  ["Student 9"],
-  ["Student 10"],
-  ["Student 11"],
-  ["Student 12"],
-];
-
 const columnWidths = [150, 50, 100, 300, 50];
 
 export default class VirtualizedDetailView extends Component {
 
+  static propTypes = {
+    section: PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        name: PropTypes.string.isRequired,
+      })).isRequired
+    }).isRequired,
+    scriptData: PropTypes.shape({
+      stages: PropTypes.arrayOf(PropTypes.shape({
+        levels: PropTypes.arrayOf(PropTypes.object).isRequired
+      })),
+      id: PropTypes.number.isRequired,
+    }).isRequired,
+  };
+
   state = {
     fixedColumnCount: 1,
     fixedRowCount: 2,
     scrollToColumn: 0,
     scrollToRow: 0,
   };
 
-  cellRenderer({columnIndex, key, rowIndex, style}) {
+  cellRenderer = ({columnIndex, key, rowIndex, style}) => {
+    const {section, scriptData} = this.props;
+
     return (
       <div className={styles.Cell} key={key} style={style}>
+        {(rowIndex === 0 && columnIndex === 0) && (
+          <span style={styles.cell}>Lesson</span>
+        )}
+        {(rowIndex === 1 && columnIndex === 0) && (
+          <span style={styles.cell}>Level Type</span>
+        )}
+        {(rowIndex >= 2 && columnIndex === 0) && (
```

Comment:
Having `students[rowIndex-2]` being calculated in multiple places feels like it'll be easy to make a mistake. Might be worth having a `const studentIndex = rowIndex - 2;` at the top so that can be used everywhere.